### DISCUSSION TITLE
Layout Help

### DISCUSSION THREAD

**Discussion by JokerMartini** [quasarframework/quasar#10684] [General - Components / Directives / etc] [answered]
I'm trying to create a layout similar to the one shown here. However I'm quite confused on how to successfully get something like this to work. I've attempted a number of solutions and options, all which have failed to do what I've drawn here. Are their any layout guru's out there that could help me achieve this look. The key here are getting the sticky areas to remain sticky and the scroll areas to have their scroll bars. Where I struggle is getting the divs to always expand to take up the vertical space remaining. The sidebar on the right can be a fixed width, however the rest is flexible to take up available space.

**Desired**
![image](https://user-images.githubusercontent.com/1416749/133434926-d607b809-f5cf-4f28-b8ba-422df8fbf478.png)

This is what I have so far. The top bar is sticky as desired, but the individual scroll bars are not displaying for the two columns, unless I hard code a fixed height for the column divs. Which in my case i want them to take as much vertical space as possible.

**DetailsPage.vue**
```
<template>
  <q-page class="flex" style="background: grey">
    <div
      class="col q-pa-md scroll"
      style="background: rgba(240, 150, 50, 0.75); height:400px;"
    >
      {{ dummyText }}
    </div>
    <div
      class="col-4 q-pa-md scroll"
      style="
        background: rgba(130, 140, 180, 0.75);
        min-width: 200px;
        max-width: 200px;
        height:400px;
      "
    >
      {{ dummyText }}
    </div>
  </q-page>
</template>
```

**Current**
![image](https://user-images.githubusercontent.com/1416749/133440421-e5df7e95-12bd-4fcd-b52a-0987f40b0a77.png)

**Comment by JokerMartini** [comment]
I got closer here, but how do i now make each column have their individual scroll bars?

```
<q-page class="row justify-start">
    <div class="row col absolute-full">
      <div class="col column">
        <!-- column content -->
        <div
          class="col q-pa-md scroll"
          style="background: rgba(240, 150, 50, 0.75)"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aut, velit
          debitis quae totam aperiam placeat aliquid odit necessitatibus, sit
          quasi ut, laborum ab facilis dolorem impedit saepe molestias fuga
          ipsa?
        </div>
      </div>
      <div class="col-4 column">
        <!-- column content -->
        <div
          class="col q-pa-md scroll"
          style="background: rgba(130, 140, 180, 0.75)"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aut, velit
          debitis quae totam aperiam placeat aliquid odit necessitatibus, sit
          quasi ut, laborum ab facilis dolorem impedit saepe molestias fuga
          ipsa?
        </div>
      </div>
    </div>
  </q-page>
```

![image](https://user-images.githubusercontent.com/1416749/133449262-a5077205-d175-485c-957d-0c49cea372d0.png)

**Comment by ontwikkelfabriek** [comment]
@JokerMartini 

> I got closer here, but how do i now make each column have their individual scroll bars?

https://quasar.dev/vue-components/scroll-area

**Reply by JokerMartini** [reply]
I did add those, how would one get the **Stick** section on the right-column in the purple bar shown in the screenshot of the original post?
```

<template>
  <q-page class="row justify-start">
    <div class="row col absolute-full">
      <div class="col column">
        <!-- column content -->
        <q-scroll-area style="height: 100%; background:green">
          <div class="q-ma-md">
            <div v-for="n in 100" :key="n" class="q-py-xs">
              Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do
              eiusmod tempor incididunt ut labore et dolore magna aliqua.
            </div>
          </div>
        </q-scroll-area>
      </div>
      <div class="col-4 column">
        <!-- column content -->
        <q-scroll-area style="height: 100%; background:red">
          <div class="q-ma-md">
            <div v-for="n in 100" :key="n" class="q-py-xs">
              Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do
              eiusmod tempor incididunt ut labore et dolore magna aliqua.
            </div>
          </div>
        </q-scroll-area>
      </div>
    </div>
  </q-page>
</template>
```

**Reply by ontwikkelfabriek** [reply]
@JokerMartini 

> stick(y)

Example:
https://quasar.dev/layout/page-sticky#expanded

**Reply by JokerMartini** [reply]
ill check that out and see. i tried messing with it and it didn't get the result i was expecting directly.

**Reply by JokerMartini** [reply]
Here it is....

```
<template>
  <!-- https://codepen.io/metalsadman/pen/ZgKexK -->
  <q-page class="row justify-start">
    <div class="row col absolute-full">
      <div class="col column">
        <!-- column content -->
        <q-scroll-area visible style="height: 100%; background: green">
          <div class="q-ma-md">
            <div v-for="n in 100" :key="n" class="q-py-xs">
              Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do
              eiusmod tempor incididunt ut labore et dolore magna aliqua.
            </div>
          </div>
        </q-scroll-area>
      </div>
      <div
        class="col-3 column no-wrap"
        style="min-width: 300px; max-width: 400px"
      >
        <!-- column content -->
        <q-card flat>
          <q-card-section> Project Title </q-card-section>
        </q-card>
        <q-scroll-area visible style="height: 100%; background: red">
          <div class="q-ma-md">
            {{ project }}
            <div v-for="n in 40" :key="n" class="q-py-xs">
              Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do
              eiusmod tempor incididunt ut labore et dolore magna aliqua.
            </div>
          </div>
        </q-scroll-area>
      </div>
    </div>
  </q-page>
</template>
```

**Comment by julie777** [comment, marked as answer]
For your layout use
```  <q-layout view="hHh lpR fFf">```

One thing that I found to be non obvious is that drawer-container and page-container are actually the entire browser window. They set margins/apply padding to leave space for the header. This is important because it changes where the scrollable area must be and it also affects the scrollbar for the entire window.

For the drawer
```    <q-drawer side="right" v-model="rightDrawerOpen" show-if-above bordered class="no-scroll">```
I needed to add no-scroll or the scrollbar will create a browser window scrollbar. This is how the entire sidebar normally scrolls.

Wrapping the section of the sidebar that I want to scroll with
```      <q-scroll-area class="full-height" visible>```
Without full-height you won't see anything. Adding the visible attribute will make the scrollbar always visible. Without it the scrollbar appears when the mouse enters the sidebar.

For the main content area "q-page" you have to put the scroll-area inside the page because the page is the full size of the browser window, but the part of the page inside the q-page is the part we want to scroll.
```    <q-scroll-area class="bg-red-1 absolute fit full-height" visible>```
Notice that for the page I also added "absolute fit" to the class. Without it you won't see the content.

I created a [codepen](https://codepen.io/Julie777/pen/YzQvYOK) so that you can see it work.
